hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/http/SolrHttpClient.java

```diff
@@ -0,0 +1,197 @@
+
+  private static CloseableHttpClient httpClient;
+  private static HttpClientContext context;
+
+  public static CloseableHttpClient getCloseableHttpClient()
+      throws KeyManagementException, NoSuchAlgorithmException,
+      KeyStoreException {
+    if (null != httpClient) {
+      return httpClient;
+    }
+    try {
+      System.setProperty("sun.security.krb5.debug", "true");
+      System.setProperty("javax.security.auth.useSubjectCredsOnly", "false");
+      Lookup<AuthSchemeProvider> authSchemeRegistry =
+          RegistryBuilder.<AuthSchemeProvider>create()
+              .register(AuthSchemes.SPNEGO, new SPNegoSchemeFactory(true))
+              .build();
+      httpClient = HttpClients.custom()
+          .setSSLHostnameVerifier(
+              NoopHostnameVerifier.INSTANCE)
+          .setSSLContext(new SSLContextBuilder().loadTrustMaterial(
+              null, new TrustStrategy() {
+                public boolean isTrusted(X509Certificate[] arg0, String arg1)
+                    throws CertificateException {
+                  return true;
+                }
+              }).build())
+          .setDefaultAuthSchemeRegistry(authSchemeRegistry).build();
+    } catch (KeyManagementException kme) {
+      LOG.error("KeyManagementException in creating http client instance", kme);
+      throw kme;
+    } catch (NoSuchAlgorithmException nsae) {
+      LOG.error("NoSuchAlgorithmException in creating http client instance",
+          nsae);
+      throw nsae;
+    } catch (KeyStoreException kse) {
+      LOG.error("KeyStoreException in creating http client instance", kse);
+      throw kse;
+    }
+    return httpClient;
+  }
+
+  private static HttpClientContext getHttpClientContext() {
+    if (null != context) {
+      return context;
+    }
+    context = HttpClientContext.create();
+    BasicCredentialsProvider credentialsProvider =
+        new BasicCredentialsProvider();
+    // This may seem odd, but specifying 'null' as principal
+    // tells java to use the logged in user's credentials
+    Credentials useJaasCreds = new Credentials() {
+      public String getPassword() {
+        return null;
+      }
+
+      public Principal getUserPrincipal() {
+        return null;
+      }
+    };
+    credentialsProvider.setCredentials(
+        new AuthScope(null, -1, null), useJaasCreds);
+    context.setCredentialsProvider(credentialsProvider);
+    return context;
+  }
+
+  public String sendPost(String host, int port,
+                         String uri, List<NameValuePair> urlParameters)
+      throws IOException {
+    String entityResponse = "";
+    String solrHostBaseQuery = getSolrHostBaseURI(host, port, uri);
+    HttpPost post = new HttpPost(solrHostBaseQuery);
+    LOG.info("Making Http Query to Solr: {} ", solrHostBaseQuery);
+    post.setEntity(new UrlEncodedFormEntity(urlParameters));
+    Instant start = Instant.now();
+    try (CloseableHttpClient httpClient = getCloseableHttpClient();
```

Comment:
httpClient is in the try source scope, it will be closed after this sendPost finish, then it will not be reused.

Reply:
> httpClient is in the try source scope, it will be closed after this sendPost finish, then it will not be reused.

Ok understood, fixed the reusability.